sources/packages/api/src/services/sfas/sfas-part-time-application.service.ts

```diff
@@ -46,6 +46,51 @@ export class SFASPartTimeApplicationsService
     await this.repo.save(application, { reload: false, transaction: false });
   }
 
+  /**
+   * Validates before an application submission to see if there is an overlapping SFAS part-time application existing.
+   * @param sin
+   * @param birthDate
+   * @param lastName
+   * @param studyStartDate
+   * @param studyEndDate
+   * @returns SFAS part-time application.
+   */
+  async validateDateOverlap(
+    sin: string,
+    birthDate: Date,
+    lastName: string,
+    studyStartDate: Date,
+    studyEndDate: Date,
+  ): Promise<SFASPartTimeApplications> {
```

Comment:
The return type can be the object or null, right? If yes we can include the null as a possible return type.

Reply:
Query is returning undefined on a happy path.

Reply:
The reason for the undefined, is due to the getOne() at the end, I feel it would be better to fetch the count of any application that are overlapping, if we are not using the application related data in the front end to show it.